iothub/service/src/DirectMethod/DirectMethodsClient.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Http2;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Subclient of <see cref="IotHubServiceClient"/> to directly invoke direct methods on devices and modules in IoT hub.
+    /// </summary>
+    /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods"/>
+    public class DirectMethodsClient
+    {
+        private const string DeviceMethodUriFormat = "/twins/{0}/methods";
+        private const string ModuleMethodUriFormat = "/twins/{0}/modules/{1}/methods";
+
+
+        private string _hostName;
+        private IotHubConnectionProperties _credentialProvider;
+        private HttpClient _httpClient;
+        private HttpRequestMessageFactory _httpRequestMessageFactory;
```

Comment:
David had me refactor these fields to be readonly like [this](https://github.com/Azure/azure-iot-sdk-csharp/blob/be46adbbe5a0a82be8b84ead85d13e1c947ef1d3/iothub/service/src/Registry2/DevicesClient.cs#L26)